plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/importapp/ImportApplicationWizard.java

```diff
@@ -44,6 +44,32 @@ public ImportApplicationWizard() {
 	}
 
 	@Override
+	public void init(IWorkbench workbench, IStructuredSelection selection) {
+		Connection connection = UIUtils.getFirstElement(selection, Connection.class);
+		if (connection != null) {
+			model.setConnection(connection);
+		} else {
+			IResource resource = UIUtils.getFirstElement(selection, IResource.class);
```

Comment:
This block is basically all doing the same thing.  Maybe a private method that tries to get the first item that is not null based on order IResource, Project, Grouping and then you have a single place where the connection is retrieved and selected item is set.  Also, it should be you will NEVER get IProject without getting IResource since a project is a resource.  You might not need that block.